### PR TITLE
Add initial animation for health bars

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on #24980 

Pretty jank and I'm not sure how to make better, or how to get it matching stable. Probably need to make a subcomponent which is operating in game time to track the interpolation correctly. Main goal was to breathe some life back into things, so maybe this is fine as a first step.

### CODE REVIEW

**Comment by bdach** [comment]
Attempted to resolve conflicts here (sorry for the force, screwed the first attempt).

File: osu.Game/Screens/Play/HUD/HealthDisplay.cs [hunk 1]
```diff
@@ -52,14 +58,46 @@ protected override void LoadComplete()
...
-            Current.BindTo(HealthProcessor.Health);
             HealthProcessor.NewJudgement += onNewJudgement;
 
+            // Don't bind directly so we can animate the startup procedure.
+            health = HealthProcessor.Health.GetBoundCopy();
+            health.BindValueChanged(h =>
+            {
+                Current.Value = h.NewValue;
+                finishInitialAnimation();
+            });
```
Comment (bdach) [inline osu.Game/Screens/Play/HUD/HealthDisplay.cs:69]: The issue with putting this in `HealthDisplay` that there's an odd one out here and it's `FailingLayer` which responsible for the red glow on the sides of the playfield. Letting it share this logic leads to it doing this:

https://github.com/ppy/osu/assets/20418176/8ddfdcb9-8022-4b0f-8766-411d5ea888eb

which probably isn't good.

Comment (peppy) [reply osu.Game/Screens/Play/HUD/HealthDisplay.cs:69]: Motherf------

I originally put this in `HealthProcessor` and moved it out thinking that the `FailingLayer` would *definitely* not be a health display.

I may end up just moving the to per-implementation. Unless you have any better ideas. In the future I can imagine individual implementations having different "startup" animations.

**Comment by bdach** [comment]
I've found and attempted to fix an issue wherein `finishInitialAnimation()` would potentially not actually properly terminate the animation, due to an in-progress `Current` bindable transform overriding changes to the health value coming from `HealthProcessor`. The fix is a bit funky but I can't find a better one, so please check it and make sure you're OK with it (and that you don't have a better alternative) before we proceed.

Looks good otherwise.

**Review by bdach — APPROVED** [review]

File: osu.Game/Screens/Play/HUD/HealthDisplay.cs [hunk 2]
```diff
@@ -107,7 +107,9 @@ private void finishInitialAnimation()
             // aside from the repeating `initialIncrease` scheduled task,
             // there may also be a `Current` transform in progress from that schedule.
             // ensure it plays out fully, to prevent changes to `Current.Value` being discarded by the ongoing transform.
-            FinishTransforms();
+            // and yes, this funky `targetMember` spec is seemingly the only way to do this
+            // (see: https://github.com/ppy/osu-framework/blob/fe2769171c6e26d1b6fdd6eb7ea8353162fe9065/osu.Framework/Graphics/Transforms/TransformBindable.cs#L21)
```
Comment (peppy) [inline osu.Game/Screens/Play/HUD/HealthDisplay.cs:111]: wow

**Comment by peppy** [comment]
The logic isn't pretty, but this is supposed to be a very quick implementation which gets built on (or replaced) so 🙈 .

**Pull request merged**